### DISCUSSION TITLE
Can't build a loadable plugin

### DISCUSSION THREAD

**Discussion by peter-jerry-ye** [WasmEdge/WasmEdge#3121] [Q&A]
Hi,

I tried to follow the steps in [the document](https://wasmedge.org/docs/contribute/plugin/develop_plugin_c).

I created a file, copied all the code on the document, built it with the command. I got a dylib file (cause I'm on mac), I moved it where the other plugins are, but when I run the code that [lists the plugins by name](https://wasmedge.org/docs/embed/c/reference/latest#plug-ins), I've only got rustls that I installed previously.

Is there any steps I'm missing?

**Comment by alabulei1** [comment]
@q82419 Could you please take a look? Thanks.

**Comment by hydai** [comment]
Hi @peter-jerry-ye 

Please provide the reproduction steps you used and the plugin code. Since several plugins on macOS are shipped to our production environment, it should not be an issue.

**Reply by peter-jerry-ye** [reply]
Sure

- I created a file named plugin.c with the content here (uploaded txt due to github limit): 
[plugin.txt](https://github.com/WasmEdge/WasmEdge/files/13790589/plugin.txt)
- I built the plugin with the command `gcc -std=c11 -DWASMEDGE_PLUGIN -lwasmedge -shared -o libwasmedgePluginTest.dylib plugin.c` (dylib cause I'm on macos)
- I moved the `libwasmedgePluginTest.dylib` to `~/.wasmedge/plugin/` where other plugins reside
- I created another file named test.c with the content here: 
[test.txt](https://github.com/WasmEdge/WasmEdge/files/13790607/test.txt)
- I built the `test.c` with the command `gcc -std=c11 -lwasmedge test.c`
- I executed it with `./a.out`

I was expecting to see the plugin I built being listed, but it only gave me
```
Number of loaded plug-ins: 1
plug-in 0 name: rustls
```

**Reply by hydai** [reply]
Hi @peter-jerry-ye 
I can reproduce this issue. Could you try the CMake way instead to see if this issue is remaining?

**Reply by peter-jerry-ye** [reply]
Yes. I tried with the following CMakeLists.txt:
```
add_library(wasmedgePluginTest
  SHARED
  plugin.c
)

set_target_properties(wasmedgePluginTest PROPERTIES
  C_STANDARD 11
)

target_compile_options(wasmedgePluginTest
  PUBLIC
  -DWASMEDGE_PLUGIN
)

target_link_libraries(wasmedgePluginTest
  PRIVATE
  wasmedge
)
```

and `cmake --build .` and copied the generated dylib file, but nothing changed. It is still not in the plugin list.

**Reply by hydai** [reply]
Let's convert this to an issue. Something goes wrong with the C. The CPP part is working, and the wasi-nn/rustls are available.
